src/utils/Utils.ts

```diff
@@ -80,3 +80,43 @@ export function getRandomValues(buffer: Uint32Array): void {
     view.setUint32(0, Math.trunc(Math.random() * 2 ** 32), true);
   }
 }
+
+/**
+ * Shim for SuppressedError
+ * https://github.com/tc39/proposal-explicit-resource-management?tab=readme-ov-file#the-suppressederror-error
+ */
+export class SuppressedError extends Error {
+  constructor(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public readonly error: any,
+    public readonly suppressed?: SuppressedError,
+    message?: string
+  ) {
+    /* istanbul ignore next -- because coverage can't detect all the branches */
+    super(message ?? `${error?.message ?? error}`);
+  }
+}
+
+/**
+ * Run a callback over the set of all values, suppress any errors, and only throw after iteration completes.
+ * @param iterable - The iterable to iterate over
+ * @param callback - The callback to run on each iteration
+ * @throws If any of the callbacks throw an error
+ */
+export function iterateEvery<T>(
+  iterable: Iterable<T> | undefined | null,
+  callback: (value: T) => void
+): void {
+  if (!iterable) return;
+  let suppressedError: SuppressedError | undefined;
+  for (const value of iterable) {
+    try {
+      callback(value);
+    } catch (err) {
+      suppressedError = new SuppressedError(err, suppressedError);
+    }
+  }
+  if (suppressedError) {
+    throw suppressedError;
```

Comment:
So essentially, before we threw fatal error at the first error encountered but now we throw a non fatal error with the last one encountered?

Reply:
yup!